### ISSUE TITLE
CodeDeploy Agent switches user after install

### ISSUE THREAD

**Issue by bayssmekanique** [aws/aws-codedeploy-agent#232]
This is occurring on the latest agent running on the Linux AMI.  I have followed the documentation provided here: [CodeDeploy Agent as non-root tutorial](https://aws.amazon.com/premiumsupport/knowledge-center/codedeploy-agent-non-root-profile/).

The `BeforeInstall` and the actual install phases both operate under the `ec2-user` as expected, but the `AfterInstall` and following commands sometimes run as `ec2-user` and sometimes run as `root`.  This makes trouble because the `pm2` instance I have setup on the machine is tied to the `ec2-user` profile because I don't want it running as `root`, but the app won't start correctly if the `ApplicationStart` phase is called by `root`.  Upwards of 90% of the time it runs as `root` and fails the deploy causing a rollback, but not 100% of the time.

For anyone else facing this issue, I have a band-aid that checks the user and performs a `sudo` to the correct user, but it's not ideal.
```sh
#!/bin/bash
# File: deployment/run.sh
if [ $(id -u) = 0 ]
then
  chmod 777 /home/ec2-user/app/deployment/run.sh
  sudo -u ec2-user -i /home/ec2-user/app/deployment/run.sh
else
  cd /home/ec2-user/app
  if [ "$DEPLOYMENT_GROUP_NAME" == "Server" ]
  then
    pm2 start pm2.config.js --env production --only metrics:server
  else
    pm2 start pm2.config.js --env production --only metrics:worker
  fi
  pm2 save
  pm2 status
fi
```

**Comment by brblck** [comment]
@bayssmekanique I was unable to reproduce this issue following the steps in our guide. 

Can you provide any steps to help reproduce the problem?
In what ways, if any, does your set differ from that of the step-by-step guide for running as non-root?

**Comment by bayssmekanique** [comment]
@brblck, I apologize but I no longer have these instances available to reference. I followed the guide I linked in original issue comment as they were presented in November of 2019, step by step, and reattempted several times across two different AMI's before I gave up. Ultimately I moved off of EC2 so I am not certain if this is a current issue or if it has been resolved.

I leave it up to you to determine if this problem sounds like one that might have been resolved since then and handle the issue as you see fit.
